From 59ed07864d3a1739b8dc1a5355057f29b9a37f78 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Fri, 9 Apr 2021 11:55:55 +0200
Subject: [PATCH] media-session: log info when loading monitor fails

Log info in all monitor plugins instead of an error.
---
 src/examples/media-session/alsa-monitor.c  | 1 +
 src/examples/media-session/bluez-monitor.c | 2 +-
 src/examples/media-session/v4l2-monitor.c  | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/examples/media-session/alsa-monitor.c b/src/examples/media-session/alsa-monitor.c
index d97a45a66..72696b772 100644
--- src/examples/media-session/alsa-monitor.c
+++ src/examples/media-session/alsa-monitor.c
@@ -1061,6 +1061,7 @@ int sm_alsa_monitor_start(struct sm_media_session *session)
 	impl->handle = pw_context_load_spa_handle(context, SPA_NAME_API_ALSA_ENUM_UDEV, NULL);
 	if (impl->handle == NULL) {
 		res = -errno;
+		pw_log_info("can't load %s: %m", SPA_NAME_API_ALSA_ENUM_UDEV);
 		goto out_free;
 	}
 
diff --git a/src/examples/media-session/bluez-monitor.c b/src/examples/media-session/bluez-monitor.c
index 9481d4472..5a4f30d48 100644
--- src/examples/media-session/bluez-monitor.c
+++ src/examples/media-session/bluez-monitor.c
@@ -586,7 +586,7 @@ int sm_bluez5_monitor_start(struct sm_media_session *session)
 	impl->handle = pw_context_load_spa_handle(context, SPA_NAME_API_BLUEZ5_ENUM_DBUS, &impl->props->dict);
 	if (impl->handle == NULL) {
 		res = -errno;
-		pw_log_error("can't load %s: %m", SPA_NAME_API_BLUEZ5_ENUM_DBUS);
+		pw_log_info("can't load %s: %m", SPA_NAME_API_BLUEZ5_ENUM_DBUS);
 		goto out_free;
 	}
 	if ((res = spa_handle_get_interface(impl->handle, SPA_TYPE_INTERFACE_Device, &iface)) < 0) {
diff --git a/src/examples/media-session/v4l2-monitor.c b/src/examples/media-session/v4l2-monitor.c
index ecb75d085..2b2a92ef3 100644
--- src/examples/media-session/v4l2-monitor.c
+++ src/examples/media-session/v4l2-monitor.c
@@ -550,6 +550,7 @@ int sm_v4l2_monitor_start(struct sm_media_session *sess)
 	impl->handle = pw_context_load_spa_handle(context, SPA_NAME_API_V4L2_ENUM_UDEV, NULL);
 	if (impl->handle == NULL) {
 		res = -errno;
+		pw_log_info("can't load %s: %m", SPA_NAME_API_V4L2_ENUM_UDEV);
 		goto out_free;
 	}
 
-- 
GitLab

